### ISSUE TITLE
Uncaught handlerTrailers exception when `decodeURI` on unencoded `grpc-message`

### ISSUE THREAD

**Issue by clww** [grpc/grpc-node#2209]
### Problem description

We are currently calling a grpc server which doesn't handle the `grpc-message` correctly, so sometimes it will send unencoded `grpc-message` containing `%`, causing the `decodeURI` ([code](https://github.com/grpc/grpc-node/blob/a56e0fc90614c2a8437c462288b6f302bee4fb8f/packages/grpc-js/src/call-stream.ts#L486)) to throw the following exception.

```
            details = decodeURI(metadataMap['grpc-message']);
                      ^

URIError: URI malformed
    at decodeURI (<anonymous>)
    at Http2CallStream.handleTrailers (.../node_modules/@grpc/grpc-js/build/src/call-stream.js:322:23)
    at ClientHttp2Stream.<anonymous> (.../node_modules/@grpc/grpc-js/build/src/call-stream.js:378:26)
    at ClientHttp2Stream.emit (node:events:520:28)
    at ClientHttp2Stream.emit (node:domain:475:12)

```

### Reproduction steps

```
decodeURI('%') <---- throws a unhandled exception, then crashes the server
```

### Additional context

There are 2 ways to handle the exception:
- catch the exception and use `endCall` to return an error ([similar code](https://github.com/grpc/grpc-node/blob/a56e0fc90614c2a8437c462288b6f302bee4fb8f/packages/grpc-js/src/call-stream.ts#L554-L562)) . or
- return the raw `grpc-message` if `decodeURI` failed

The [gRPC HTTP/2 spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses) seems to prefer the second way:

> When decoding invalid values, implementations MUST NOT error or throw away the message. At worst, the implementation can abort decoding the status message altogether such that the user would received the raw percent-encoded form. Alternatively, the implementation can decode valid portions while leaving broken %-encodings as-is or replacing them with a replacement character (e.g., '?' or the Unicode replacement character).

**Comment by murgatroid99** [comment]
I have published a fix for this in version 1.6.11

**Comment by clww** [comment]
Thanks @murgatroid99 ! though I found that there is a typo in the PR changes. The `grpc-messages` in the catch block should be `grpc-message`.
